domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/pathAssertions.kt

```diff
@@ -395,5 +396,8 @@ fun <T : Path> _parent(expect: Expect<T>): ExtractedFeaturePostStep<T, Path> =
         .withoutOptions()
         .build()
 
+fun <T : Path, String, R: Path> _resolve(expect: Expect<T>, f: KFunction2<T, String, R>, other: String): ExtractedFeaturePostStep<T, Path> =
```

Comment:
don't expect the KFunction here, using Path.resolve should be internal to this function.
```suggestion
fun <T : Path> _resolve(expect: Expect<T>, other: String): ExtractedFeaturePostStep<T, Path> =
```
Also your definition `fun <T: Path, String, R: Path>` actually defines a type parameter named `String`. That's certainly not what you want :slightly_smiling_face:  Most likely the reason why you run into type inference problems. Note also, that you don't require `R` either as you don't use it afterwards.